compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/useEffect-nested-lambdas.expect.md

```diff
@@ -51,7 +51,8 @@ function Component(props) {
   }
   const exit = t0;
   let t1;
-  if ($[2] !== exit || $[3] !== item.value) {
```

Comment:
this presumably change auto deps inference too for effects. this one is interesting, cc @jbrown215

Reply:
`item` should be deeply-immutable because it is returned from a hook. Shouldn't it be safe to take item.value as a dep?

Reply:
it's not guaranteed to be non-null